src/titiler/core/titiler/core/factory.py

```diff
@@ -166,6 +177,32 @@ def url_for(self, request: Request, name: str, **path_params: Any) -> str:
             base_url += self.router_prefix.lstrip("/")
         return url_path.make_absolute_url(base_url=base_url)
 
+    def add_route_dependencies(
+        self,
+        scopes: List[EndpointScope],
+        dependencies=List[params.Depends],
+    ):
+        """Add dependencies to routes.
+
+        Allows a developer to add dependencies to a route after the route has been defined.
+
+        """
+        for route in self.router.routes:
+            for scope in scopes:
+                match, _ = route.matches({"type": "http", **scope})
+                if match != Match.FULL:
+                    continue
+
+                # Mimicking how APIRoute handles dependencies:
+                # https://github.com/tiangolo/fastapi/blob/1760da0efa55585c19835d81afa8ca386036c325/fastapi/routing.py#L408-L412
+                for depends in dependencies[::-1]:
+                    route.dependant.dependencies.insert(  # type: ignore
+                        0,
+                        get_parameterless_sub_dependant(
+                            depends=depends, path=route.path_format  # type: ignore
+                        ),
+                    )
```

Comment:
From https://github.com/stac-utils/stac-fastapi/pull/295/commits/99445db5a99db61273dd6993dffb500eee18d18d

```suggestion
                    )

            # Register dependency on route so that it isn't ignored if the routes
            # are added to an apt after this is route (e.g. app.include_router(router))
            # https://github.com/tiangolo/fastapi/blob/58ab733f19846b4875c5b79bfb1f4d1cb7f4823f/fastapi/routing.py#L677-L678
            route.dependencies.extend(dependencies)

```